### PR TITLE
Content: HTML W4  - components

### PR DESCRIPTION
**Pull request by SallyMcGrath**
## What does this change?

Module: HTML
Week(s): 4

## Checklist

- [x] I have read the [contributing guidelines](CONTRIBUTING.MD)
- [x] I have checked my spelling and grammar with an [automated tool](https://www.grammarly.com/grammar-check)
- [x] I have previewed my changes to check the [markdown renders](https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax) as I intend
- [x] I have run my code to check it works CAVEAT EXPECTS WORKSHOP TO BE MERGED PLEASE
- [x] My changes follow our [Style Guide](https://curriculum.codeyourfuture.io/guides/code-style-guide)

## Description

w4, which previously was basically empty, now has a little workshop on components and spends some more time practising Git in class.  This git exercise expects several prior activities with merging stuff that isn't actual code to have happened in the previous 7 weeks.

The in-class git and documentation practice replaces the previous in-class practice of shipping to Netlify, as we do that in ITD and FUN now 

Only 135 minutes of study group as the PD exercises take up almost the whole morning, which we might need to push back on.

### CODE REVIEW

**Comment by Dedekind561** [comment]
I'm going to set aside some time to review this PR tomorrow!
`new Promise()`

**Comment by SallyMcGrath** [comment]
> I'm going to set aside some time to review this PR tomorrow! `new Promise()`

I do need to merge this soon please someone

**Review by 40thieves — APPROVED** [review]
Some extremely minor suggestions, but happy to merge this :shipit:

**Review by illicitonion — APPROVED** [review]
LGTM! I made a small PR here also: https://github.com/CodeYourFuture/CYF-Workshops/pull/109

File: content/html-css/prep/index.md [hunk 1]
```diff
@@ -8,7 +8,30 @@ weight = 1
 backlog= 'Module-HTML-CSS'
 +++
 
-## What should we do first?
+## 🧰 Check your setup
+
+- [ ] You must have a UNIX operating system (MacOS or Linux). Don't use Windows.
```
[the remark this answers is not in the document]
Comment (SallyMcGrath) [reply content/html-css/prep/index.md:13]: Yes I used to have a long explanation/justification on the docs but it in no way stopped people asking about it, so I gave up. I need my energy for more important debates, tbh!

File: content/html-css/prep/index.md [hunk 2]
```diff
@@ -8,7 +8,30 @@ weight = 1
...
+{{<note type="tip" title="VS Code format on save">}}
+
+1. In Visual Studio open the settings file (Code > Preferences > Settings)
+1. Search for editor format
+1. Set `editor.formatOnSave` and `editor.formatOnPaste` to `true`
+1. Set the "Default Formatter" to be Prettier
+1. In your editor, write some html and save the file. It should be formatted (indented) automatically.
+   {{</note>}}
+
+Ask for help on Slack if you are having trouble with Prettier. If a group of you are blocked by this, schedule thirty minutes of your class study time to set this up together.
```
[the remark this answers is not in the document]
Comment (SallyMcGrath) [reply content/html-css/prep/index.md:30]: Hm, no, that hasn't come up lately. The Joining Of The Slack happens in the PD work I think. (I will check with K and L)

File: content/html-css/blocks/what-are-components/index.md [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+#### 🚅 Efficiency
+
+We can split up the tasks and work in parallel. One person can work on the button component, another on the card component. When a new developer joins the team, they can make changes to a limited part of the codebase, rather than having to understand the whole thing first.
+
+</details>
+
+<details>
+<summary>How do we split up HTML and CSS into composable pieces?</summary>
+
+So, we actually can't yet, not really! We are only beginning our journey with components now. (If you are here from a scheduled review, update your answer.)
```
[the remark this answers is not in the document]
Comment (SallyMcGrath) [reply content/html-css/blocks/what-are-components/index.md:62]: (I'll redo this with web components for JS2.)

File: content/html-css/prep/index.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (40thieves) [reply content/html-css/prep/index.md:13]: Right, fair enough 👍

File: content/html-css/blocks/what-are-components/index.md [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+If we call a header component, we know that _all_ our headers will match on our site, because they all use the same shared rules.
+
+#### 🍱 Composition
+
+We can build bigger components out of smaller components and compose them onto pages to make views. This is how we build complicated interfaces quickly.
+
+#### 🚅 Efficiency
+
+We can split up the tasks and work in parallel. One person can work on the button component, another on the card component. When a new developer joins the team, they can make changes to a limited part of the codebase, rather than having to understand the whole thing first.
+
```
Comment (SallyMcGrath) [inline content/html-css/blocks/what-are-components/index.md:56]: #### 🦻 Communication

All these are really about communication. With components, we create a shared design _language_ to describe the application we are building. Developers can use this language with each other, and also share parts of it with non-developers, like designers and customers. 

We learned in Fundamentals that much of the work of developers is in _interpreting_ the requirements of customers and expressing those requirements, first in specifications like user stories and then again in computer languages. 

We also need to learn to express _our_ requirements to non-coders, by organising the complexities of users' needs into small, comprehensible blocks. Components help us do this.

File: content/html-css/prep/index.md (hunk 2 above)
Comment (SallyMcGrath) [inline content/html-css/prep/index.md:30]:
```suggestion
Ask for help on main Slack if you are having trouble with Prettier. If a group of you are blocked by this, schedule thirty minutes of your class study time to set this up together.
```

**Review by SallyMcGrath — COMMENTED** [review]
short edit for main Slack

**Pull request merged**